src/main/java/net/fabricmc/fabric/mixin/client/model/MixinBakedQuad.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.model;
+
+import org.spongepowered.asm.mixin.Mixin;
+
+import net.fabricmc.fabric.api.client.model.FabricBakedQuad;
+import net.fabricmc.fabric.api.client.model.FabricVertexFormat;
+import net.minecraft.client.render.model.BakedQuad;
+
+/**
+ * Causes all BakedQuad instances to also implement FabricBakedQuad.
+ * Purely additive - makes no change to internals or existing method contracts.
+ */
+@Mixin(BakedQuad.class)
+public abstract class MixinBakedQuad implements FabricBakedQuad {
```

Comment:
Random idea: Add a way to mark BakedQuads (and/or FabricBakedQuads?) as immutable. Then, make sure BakedQuads produced by Minecraft's loading process are all marked as immutable.

Reply:
This is a great idea.  I do anticipate mutable instances of FabricBakedQuads. Nothing in the spec prevents it and it would be useful in many cases.  Nothing in the current API really makes use of it but it will encourage authors to think about consequences and may save people a lot of grief down the road.

Reply:
There were a few mods which had mutable instances of BakedQuads in 1.8~1.12, and it prevented me from doing further-reaching optimizations with model deduplications. As such, make mutability the assumed default, but allow mods to mark their quad classes as immutable.

Reply:
I made immutability the default before I read this.  I suppose that makes some generous assumptions that authors will pay attention.  Does it need changed?

Reply:
**YES.**

Reply:
Sorry for bolding it so much, but let's just say I've been surprised during FoamFix development one too many times...

Reply:
Not a problem at all.  Will be in next push.